### PR TITLE
links in the docs,reviewed 

### PR DESCRIPTION
**Pull request by rahul799**
Hello! Please make sure to check all these boxes before submitting a Pull Request
(PR). Once you have checked the boxes, feel free to remove all text except the
justification in point 4. 

1. [x] You have run tests on this submission, either by using [Travis Continuous Integration testing](https://github.com/pysal/pysal/wiki/GitHub-Standard-Operating-Procedures#automated-testing-w-travis-ci) testing or running `nosetests` on your changes?
2. [x] This pull introduces new functionality covered by [docstrings](https://en.wikipedia.org/wiki/Docstring#Python) and [unittests](https://docs.python.org/3.6/library/unittest.html)?
3. [x] You have [assigned a reviewer](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) and added relevant [labels](https://help.github.com/articles/applying-labels-to-issues-and-pull-requests/).
4. [x] The justification for this PR is:
    * It may fix the issue #338

### CODE REVIEW

**Comment by jGaboardi** [comment]
#338 #337 #336

**Comment by rahul799** [comment]
> @rahul799 I would like to make sure that the `html` files were generated with instructions set forth in the [Publishing the docs with Github Page](http://pysal.org/getting_started#publishing-the-docs-with-github-page) section of [Getting Started](http://pysal.org/getting_started). Is this correct?

@jGaboardi I don't think that it will voilate the [Getting Started](http://pysal.org/getting_started), as I have just shifted the link to a [new address](https://pysal.org/libpysal/generated/libpysal.weights.W.html#libpysal.weights.W). But am not sure.

**Comment by jGaboardi** [comment]
> @jGaboardi I don't think that it will voilate the Getting Started, as I have just shifted the link to a new address. But am not sure.

It's necessarily about violating procedure, it's that the `html` files in the documentation are generated through the method detailed in [Publishing the docs with Github Page](http://pysal.org/getting_started#publishing-the-docs-with-github-page). It is best practice to follow these guidelines.

**Comment by rahul799** [comment]
Thanks for putting some light on this topic. I'll consider your guidance and follow the procedures accordingly.

File: docsrc/Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # You can set these variables from the command line.
 SPHINXOPTS    =
 SPHINXBUILD   = sphinx-build
-SPHINXPROJ    = spaghetti
+SPHINXPROJ    = PACKAGE_NAME
```
Comment (weikang9009) [inline docsrc/Makefile]: This should not be changed?

Comment (jGaboardi) [reply docsrc/Makefile]: Correct. @rahul799 the `PACKAGE_NAME` variable is used in the [submodule_template](https://github.com/pysal/submodule_template/blob/master/doc/Makefile) in reference to what the package name actually is. Here, this should remain `spaghetti`.

Comment (rahul799) [reply docsrc/Makefile]: Ya, am sorry, actually I downloaded the new Makefile from [submodile template](https://github.com/pysal/submodule_template/blob/master/docsrc/Makefile) & forgot to change the Package_Name here. Thanks @weikang9009, @jGaboardi for pointing this out.

**Comment by jGaboardi** [comment]
@rahul799 can you turn on [`allow edits from maintainers`](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for me? I would like to push up some commits to this PR.

**Comment by jGaboardi** [comment]
@knaaptime @weikang9009 

I am having a difficult time getting the `intersphinx_mapping` to work in the `spaghetti` docs (I am working locally with [`rahul's branch`](https://github.com/rahul799/spaghetti/tree/bugfix)). The docs are building but I am getting a `RemovedInSphinx30Warning`—[though it seems to not be the culprit](https://stackoverflow.com/questions/57401254/fix-sphinx-removedinsphinx30warning) (full output below). Since the `geosnap` docs are building correctly (i.e. [`geosnap.analyze.cluster_spatial`](https://spatialucr.github.io/geosnap/generated/geosnap.analyze.cluster_spatial.html#geosnap.analyze.cluster_spatial) I have attempted to mimic the [`geosnap/docsrc/conf.py`](https://github.com/spatialucr/geosnap/blob/master/docsrc/conf.py) file, but I am still getting the same results.

Any input would be greatly appreciated.

```
user:docsrc user$ make html
Running Sphinx v2.3.0
loading pickled environment... failed
failed: No such config value: user_agent
[autosummary] generating autosummary for: api.rst, generated/spaghetti.Network.NetworkF.rst, generated/spaghetti.Network.NetworkG.rst, generated/spaghetti.Network.NetworkK.rst, generated/spaghetti.Network._evaluate_napts.rst, generated/spaghetti.Network._extractnetwork.rst, generated/spaghetti.Network._newpoint_coords.rst, generated/spaghetti.Network._round_sig.rst, generated/spaghetti.Network._snap_to_link.rst, generated/spaghetti.Network._yield_napts.rst, ..., generated/spaghetti.Network.savenetwork.rst, generated/spaghetti.Network.simulate_observations.rst, generated/spaghetti.Network.snapobservations.rst, generated/spaghetti.Network.split_arcs.rst, generated/spaghetti.PointPattern.rst, generated/spaghetti.SimulatedPointPattern.rst, generated/spaghetti.element_as_gdf.rst, index.rst, installation.rst, references.rst
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://docs.scipy.org/doc/numpy/objects.inv...
loading intersphinx inventory from https://pandas.pydata.org/pandas-docs/stable/objects.inv...
loading intersphinx inventory from https://geopandas.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://pysal.org/libpysal/objects.inv...
loading intersphinx inventory from https://esda.readthedocs.io/en/latest/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 33 source files that are out of date
updating environment: [new config] 33 added, 0 changed, 0 removed
checking for /Users/user/spaghetti/docsrc/_static/references.bib in bibtex cache... not found                                                   
parsing bibtex file /Users/user/spaghetti/docsrc/_static/references.bib... parsed 2 entries

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
/Users/user/miniconda3/envs/py3_spgh_dev/lib/python3.7/site-packages/sphinx_bootstrap_theme/bootstrap/layout.html:41: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
  </div>
writing output... [100%] references                                                                                                                  
generating indices...  genindexdone
highlighting module code... [100%] spaghetti.network                                                                                                 
writing additional pages...  search/Users/user/miniconda3/envs/py3_spgh_dev/lib/python3.7/site-packages/sphinx_bootstrap_theme/bootstrap/search.html:20: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
  {{ super() }}
done
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in _build/html.
```

**Comment by knaaptime** [comment]
the mapping is correct, you just need to tun on crossreferencing in your conf.py: `numpydoc_xref_param_type = True`

**Comment by jGaboardi** [comment]
> the mapping is correct, you just need to tun on crossreferencing in your conf.py: `numpydoc_xref_param_type = True`

Yeah, I have that in my local branch and still not producing links within the `html` files. 😕 . Completely lost as to what I am doing wrong.

**Comment by knaaptime** [comment]
huh. i tested before i commented and it works on mine..?
image

what version of sphinx are you using?

**Comment by jGaboardi** [comment]
Well, heck... it's obviously something either (a) I am doing incorrectly; or (b) wrong with my machine/configuration. The following of `spaghetti.Network.snapobservations` is what I have locally when generated with either `sphinx v2.2.2` or `v2.3.0`:

Screenshot 2019-12-18 11 37 50

**Comment by jGaboardi** [comment]
@knaaptime seems like there was some corruption in my install of `sphinx`. After removing everything tied to `sphinx` and re-installing, I can produce the `intersphinx_mapping` as desired!

**Comment by jGaboardi** [comment]
@rahul799 I need to make corrections and update the `spaghetti` docs in order to meet a release deadline (12/31). If you do not have the time to work on this PR, it is not a problem. I will open up another PR for this issue.

**Comment by rahul799** [comment]
@jGaboardi It wIll be better if you open another PR. As am not able to work on this due to disruption of internet connection because of some emergency situations in the state(india).

**Comment by jGaboardi** [comment]
@rahul799 That is understandable. I would still like you to get credit for opening the initial PR, so would you be able to turn on [`allow edits from maintainers`](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)? That way I can add to this PR. As a last resort I will open a new PR.

**Comment by rahul799** [comment]
> @rahul799 That is understandable. I would still like you to get credit for opening the initial PR, so would you be able to turn on [`allow edits from maintainers`](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)? That way I can add to this PR. As a last resort I will open a new PR.

@jGaboardi Thankyou for understanding. I have turned on `allow edits from maintainers`.You can open the new PR.

**Comment by jGaboardi** [comment]
@rahul799, thanks so much for starting this PR. I am closing it for now and it is being continued in #342! Please feel free to contribute again!

**Comment by rahul799** [comment]
@jGaboardi I am very glad to contribute to such a great organisation. Meanwhile I was not able to continue due to some **[unavoidable circumstances](https://www.livemint.com/news/india/caa-protests-internet-suspended-in-lucknow-ghaziabad-and-20-districts-in-up-today-11577419226284.html)** but I will really like to contribute again.

**Comment by jGaboardi** [comment]
@rahul799 oh geez, that is unfortunate... Your contributions are welcome here any time!

**Pull request closed without merging**